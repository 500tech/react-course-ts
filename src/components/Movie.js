import * as React from 'react';

const Movie = ({ label }) => <li>{label}</li>;

export default Movie;
