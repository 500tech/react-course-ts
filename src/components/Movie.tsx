import * as React from 'react';

interface ComponentProps {
  movie: {
    label: string,
    rating: number,
    id: number
  },
  selectMovie: (id) => void,
  removeMovie: (id) => void
}

const Movie = (props: ComponentProps) => (
  <li onClick={() => props.selectMovie(props.movie.id)}>
    <div>{props.movie.label}</div>
    <div
      className="remove"
      onClick={() => props.removeMovie(props.movie.id)}>
      remove
    </div>
  </li>
);

export default Movie;
