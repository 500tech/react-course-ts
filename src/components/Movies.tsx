import * as React from 'react';
import Button from './common/Button';
import Input from './common/Input';
import Movie from './Movie';

interface ComponentProps {
  data: Array<{
    label: string,
    rating: number,
    id: number
  }>,
  addMovie: () => void,
  clearAll: () => void,
  removeMovie: (id) => void,
  selectMovie: (id) => void,
  updateValue: (val) => void,
  inputVal: string
}

const Movies = (props: ComponentProps) => (
  <div className="movies">
    <div className="header">my favorite movies</div>

    <Input onChange={props.updateValue} value={props.inputVal} />

    <Button onClick={props.addMovie}>add movie</Button>

    {
      props.data.length ? (
        <ul>
          {
            props.data.map(movie => (
              <Movie
                key={movie.id}
                movie={movie}
                selectMovie={props.selectMovie}
                removeMovie={props.removeMovie}
              />
            ))
          }
        </ul>
      ) : (
        <div className="empty">
          no movies
        </div>
      )
    }

    <div className="filters">
      <div className="clear" onClick={props.clearAll}>clear all</div>
    </div>
  </div>
);

export default Movies;
