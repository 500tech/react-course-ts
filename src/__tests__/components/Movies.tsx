import * as React from 'react';
import { shallow } from 'enzyme';
import { noop } from 'lodash';
import Movies from "../../components/Movies";
import Movie from "../../components/Movie";

const setup = ({
  data = [],
  addMovie = noop,
  clearAll = noop,
  removeMovie = noop,
  selectMovie = noop,
  updateValue = noop,
  inputVal = ''
} = {}) => ({
  data,
  addMovie,
  clearAll,
  selectMovie,
  removeMovie,
  updateValue,
  inputVal
});

describe('<Movies/>', () => {

  const movies = [
    { label: 'test movie', rating: 1, id: 1 },
    { label: 'test movie 2', rating: 2, id: 2 },
    { label: 'test movie 3', rating: 3, id: 3 }
  ];

  test('should render without movies', () => {
    const props = setup();
    const component = shallow(
        <Movies {...props} />
    );

    expect(component).toMatchSnapshot();
    expect(component.find('.empty')).toExist();
  });

  test('should render with movies', () => {
    const props = setup({ data: movies });
    const component = shallow(
        <Movies {...props} />
    );

    expect(component).toMatchSnapshot();
    expect(component.find(Movie).length).toEqual(3);
  });

  test('should fire clearAll when clicking on clear all', () => {
    const props = setup({ clearAll: jest.fn() });
    const component = shallow(
      <Movies {...props} />
    );

    expect(props.clearAll).not.toHaveBeenCalled();

    component.find('.clear').simulate('click');

    expect(props.clearAll).toHaveBeenCalled();
  });
});
