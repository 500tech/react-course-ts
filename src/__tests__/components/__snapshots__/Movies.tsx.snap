// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movies/> should render with movies 1`] = `
<div
  className="movies"
>
  <div
    className="header"
  >
    my favorite movies
  </div>
  <Input
    onChange={[Function]}
    value=""
  />
  <Component
    onClick={[Function]}
  >
    add movie
  </Component>
  <ul>
    <Movie
      key="1"
      movie={
        Object {
          "id": 1,
          "label": "test movie",
          "rating": 1,
        }
      }
      removeMovie={[Function]}
      selectMovie={[Function]}
    />
    <Movie
      key="2"
      movie={
        Object {
          "id": 2,
          "label": "test movie 2",
          "rating": 2,
        }
      }
      removeMovie={[Function]}
      selectMovie={[Function]}
    />
    <Movie
      key="3"
      movie={
        Object {
          "id": 3,
          "label": "test movie 3",
          "rating": 3,
        }
      }
      removeMovie={[Function]}
      selectMovie={[Function]}
    />
  </ul>
  <div
    className="filters"
  >
    <div
      className="clear"
      onClick={[Function]}
    >
      clear all
    </div>
  </div>
</div>
`;

exports[`<Movies/> should render without movies 1`] = `
<div
  className="movies"
>
  <div
    className="header"
  >
    my favorite movies
  </div>
  <Input
    onChange={[Function]}
    value=""
  />
  <Component
    onClick={[Function]}
  >
    add movie
  </Component>
  <div
    className="empty"
  >
    no movies
  </div>
  <div
    className="filters"
  >
    <div
      className="clear"
      onClick={[Function]}
    >
      clear all
    </div>
  </div>
</div>
`;
